### ISSUE TITLE
Add snappy encoding support in RequestDecompressionLayer

### ISSUE THREAD

**Issue by nullren** [tower-rs/tower-http#529]
- [x] I have looked for existing issues (including closed) about this

### Motivation

Add `snappy` as one of the encodings to decode in RequestDecompressionLayer. It is fairly commonly used, eg, it is required for Prometheus remote-write.

### Proposal

Using the [snap](https://crates.io/crates/snap) crate, add a new `decompression-snappy` feature similar to the other encodings.

### Alternatives

Did not consider alternatives.

**Comment by jplatte** [comment]
Is this standardized, or in the process of being standardized? MDN has no info on it [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding).

**Comment by jplatte** [comment]
No, I don't want to support non-standard extensions. This crate is already undermaintained and this would almost certainly make things worse. I would recommend you make your own crate, maybe starting from `tower-http`s `compression` module. If it ends up working well and being well-maintained, maybe we could even deprecate tower-http's `compression` module in the future and point to the alternative crate.
